src/libmongoc/tests/test-mongoc-sdam-monitoring.c

```diff
@@ -958,6 +958,241 @@ test_no_duplicates (void)
+      t->pool = test_framework_client_pool_new_from_uri (uri, NULL);
+      test_framework_set_pool_ssl_opts (t->pool);
+      mongoc_uri_destroy (uri);
+   }
+
+   // Capture heartbeat events
+   {
+      mongoc_apm_callbacks_t *cbs = mongoc_apm_callbacks_new ();
+      mongoc_apm_set_server_heartbeat_started_cb (cbs, heartbeat_started);
+      mongoc_apm_set_server_heartbeat_succeeded_cb (cbs, heartbeat_succeeded);
+      mongoc_apm_set_server_heartbeat_failed_cb (cbs, heartbeat_failed);
+      mongoc_client_pool_set_apm_callbacks (t->pool, cbs, t);
+      mongoc_apm_callbacks_destroy (cbs);
+   }
+
+   // Pop and push a client to start monitoring
+   {
+      mongoc_client_t *client = mongoc_client_pool_pop (t->pool);
+      mongoc_client_pool_push (t->pool, client);
+   }
+
+   return t;
+}
+
+// `smm_wait` waits for `count` heartbeat events.
+static bool
+smm_wait (smm_t *t, size_t count)
+{
+   int64_t started = bson_get_monotonic_time ();
+   while (true) {
+      bson_mutex_lock (&t->lock);
+      if (t->events_len >= count) {
+         bson_mutex_unlock (&t->lock);
+         return true;
+      }
+      bson_mutex_unlock (&t->lock);
+
+      int64_t now = bson_get_monotonic_time ();
+      // Timeout
+      if (now - started > 10 * 1000 * 1000) {
+         break;
+      }
+      _mongoc_usleep (500 * 1000); // Sleep for 500ms.
+   }
+   return false;
+}
+
+
+static void
+smm_assert (smm_t *t, bool stream)
+{
+   bson_mutex_lock (&t->lock);
+
+   // First two events should always be a non-awaited heartbeat started and succeeded
+   ASSERT_CMPSIZE_T (t->events_len, ==, 3);
+   ASSERT_CMPINT (t->events[0].type, ==, SERVER_HEARTBEAT_STARTED);
+   ASSERT (!t->events[0].awaited);
+   ASSERT_CMPINT (t->events[1].type, ==, SERVER_HEARTBEAT_SUCCEEDED);
+   ASSERT (!t->events[1].awaited);
+   ASSERT_CMPINT (t->events[2].type, ==, SERVER_HEARTBEAT_STARTED);
+   ASSERT (stream == t->events[2].awaited); // check for stream or pool
+   bson_mutex_unlock (&t->lock);
+}
+
+static void
+smm_destroy (smm_t *t)
+{
+   if (!t) {
+      return;
+   }
+   mongoc_client_pool_destroy (t->pool);
+   bson_mutex_destroy (&t->lock);
+   bson_free (t);
+}
+
+// `test_serverMonitoringMode` implements spec tests from serverMonitoringMode.yml
+// The spec test needs a client pool for stream monitoring. The unified test runner only uses single-threaded
+// clients.
+static void
+test_serverMonitoringMode (void)
+{
+   if (test_framework_is_replset ()) {
+      printf ("Test is skipped. SDAM events are non-deterministic when monitoring multiple servers.");
+      return;
+   }
+
+   smm_t *t = NULL;
+   mongoc_handshake_t *md = _mongoc_handshake_get ();
+
+   if (test_framework_get_server_version () >= test_framework_str_to_version ("4.4.0")) {
+      printf ("'connect with serverMonitoringMode=auto >=4.4' ... begin\n");
+      t = smm_new ("auto");
+      ASSERT (smm_wait (t, 3));
+      {
+         smm_assert (t, true);
+      }
```

Comment:
Remove unnecessary scope.

```suggestion
      smm_assert (t, true);
```